### PR TITLE
Added the spot endpoint

### PR DESCRIPTION
**Pull request by adambrodin**
Added the spot endpoint and all it's functions - /GET /POST /PUT /DELETE

### CODE REVIEW

File: Source/LinuxParking.API/Controllers/SpotController.cs [hunk 1]
```diff
@@ -1,49 +1,108 @@
...
+            _spotService = spotService;
+            _mapper = mapper;
         }
 
         [HttpPost]
-        public async Task CreateAsync([FromRoute] string stationId)
+        public async Task<IActionResult> CreateAsync([FromRoute] int stationId, [FromBody] CreateSpotResource resource)
         {
-            throw new NotImplementedException();
+            if (!ModelState.IsValid)
```
Comment (TheNerdyHamster) [inline Source/LinuxParking.API/Controllers/SpotController.cs:27]: We should follow the same "style" trough the whole project.
If you could change this to:
```csharp
if (stmt)
    return <value>
```

File: Source/LinuxParking.API/Controllers/SpotController.cs [hunk 2]
```diff
@@ -1,49 +1,108 @@
...
+            {
+                return BadRequest(ModelState.GetErrorMessages());
+            }
+
+            var spot = _mapper.Map<CreateSpotResource, Spot>(resource);
+            spot.StationId = stationId;
+            var res = await _spotService.SaveAsync(spot);
+
+            if (!res.Success)
+            {
```
Comment (TheNerdyHamster) [inline Source/LinuxParking.API/Controllers/SpotController.cs]: We should follow the same "style" trough the whole project.
If you could change this to:
```csharp
if (stmt)
    return <value>
```

File: Source/LinuxParking.API/Controllers/SpotController.cs [hunk 3]
```diff
@@ -1,49 +1,108 @@
...
         }
 
         [HttpGet]
-        public async Task GetAllAsync([FromRoute] string stationId)
+        public async Task<IActionResult> GetAllAsync([FromRoute] int stationId)
         {
-            throw new NotImplementedException();
+            var res = await _spotService.ListAsync(stationId);
+
+            if (!res.Success)
```
Comment (TheNerdyHamster) [inline Source/LinuxParking.API/Controllers/SpotController.cs:46]: We should follow the same "style" trough the whole project.
If you could change this to:
```csharp
if (stmt)
    return <value>
```

File: Source/LinuxParking.API/Controllers/SpotController.cs [hunk 4]
```diff
@@ -1,49 +1,108 @@
...
         }
 
         [HttpGet]
         [Route("{spotId}")]
-        public async Task GetAsync([FromRoute] string stationId, [FromRoute] string spotId)
+        public async Task<IActionResult> GetAsync([FromRoute] int stationId, [FromRoute] int spotId)
         {
-            throw new NotImplementedException();
+            var res = await _spotService.FindByIdAsync(stationId, spotId);
+            if (!res.Success)
```
Comment (TheNerdyHamster) [inline Source/LinuxParking.API/Controllers/SpotController.cs:58]: We should follow the same "style" trough the whole project.
If you could change this to:
```csharp
if (stmt)
    return <value>
```

File: Source/LinuxParking.API/Controllers/SpotController.cs [hunk 5]
```diff
@@ -1,49 +1,108 @@
...
         }
 
         [HttpPut]
         [Route("{spotId}")]
-        public async Task UpdateAsync([FromRoute] string stationId, [FromRoute] string spotId)
+        public async Task<IActionResult> UpdateAsync(int stationId, int spotId, [FromBody] CreateSpotResource resource)
         {
-            throw new NotImplementedException();
+            if (!ModelState.IsValid)
+            {
```
Comment (TheNerdyHamster) [inline Source/LinuxParking.API/Controllers/SpotController.cs]: We should follow the same "style" trough the whole project.
If you could change this to:
```csharp
if (stmt)
    return <value>
```

File: Source/LinuxParking.API/Controllers/SpotController.cs [hunk 6]
```diff
@@ -1,49 +1,108 @@
...
+                return BadRequest(ModelState.GetErrorMessages());
+            }
+
+            var spot = _mapper.Map<CreateSpotResource, Spot>(resource);
+            var res = await _spotService.UpdateAsync(stationId, spotId, spot);
+
+            if (!res.Success)
```
Comment (TheNerdyHamster) [inline Source/LinuxParking.API/Controllers/SpotController.cs:75]: We should follow the same "style" trough the whole project.
If you could change this to:
```csharp
if (stmt)
    return <value>
```

File: Source/LinuxParking.API/Controllers/SpotController.cs [hunk 7]
```diff
@@ -1,49 +1,108 @@
...
+        public async Task<IActionResult> UpdateAsync(int stationId, int spotId, [FromBody] CreateSpotResource resource)
         {
-            throw new NotImplementedException();
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState.GetErrorMessages());
+            }
+
+            var spot = _mapper.Map<CreateSpotResource, Spot>(resource);
+            var res = await _spotService.UpdateAsync(stationId, spotId, spot);
```
Comment (TheNerdyHamster) [inline Source/LinuxParking.API/Controllers/SpotController.cs:73]: You could map spotId, to the `spot` object

File: Source/LinuxParking.API/Domain/Resources/CreateSpotResource.cs [hunk 8]
```diff
@@ -0,0 +1,12 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace LinuxParking.API.Domain.Resources
+{
+    public class CreateSpotResource
+    {
+        [Required]
+        public int Size { get; set; }
```
Comment (TheNerdyHamster) [inline Source/LinuxParking.API/Domain/Resources/CreateSpotResource.cs:9]: Might be worth setting a attribute rule so size cant go under 1

File: Source/LinuxParking.API/Domain/Resources/CreateSpotResource.cs [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+        [Required]
+        public decimal Price { get; set; }
```
Comment (TheNerdyHamster) [inline Source/LinuxParking.API/Domain/Resources/CreateSpotResource.cs:12]: Same as the size property

**Review by TheNerdyHamster — CHANGES_REQUESTED** [review]
Checked over, everything looks good. Commented on a few things that might need a change

File: Source/LinuxParking.API/Controllers/SpotController.cs (hunk 7 above)
Comment (adambrodin) [reply Source/LinuxParking.API/Controllers/SpotController.cs:73]: Dismissed.

File: Source/LinuxParking.API/Domain/Resources/CreateSpotResource.cs [hunk 10]
```diff
@@ -5,8 +5,10 @@ namespace LinuxParking.API.Domain.Resources
     public class CreateSpotResource
     {
         [Required]
+        [MinLength(1)]
         public int Size { get; set; }
         [Required]
+        [MinLength(1)]
```
Comment (TheNerdyHamster) [inline Source/LinuxParking.API/Domain/Resources/CreateSpotResource.cs]: Minlength will be valid if length is 0.

File: Source/LinuxParking.API/Domain/Resources/CreateSpotResource.cs [hunk 11]
```diff
@@ -5,8 +5,10 @@ namespace LinuxParking.API.Domain.Resources
     public class CreateSpotResource
     {
         [Required]
+        [MinLength(1)]
```
Comment (TheNerdyHamster) [inline Source/LinuxParking.API/Domain/Resources/CreateSpotResource.cs]: Minlength will be valid if price is 0.

**Review by TheNerdyHamster — CHANGES_REQUESTED** [review]

**Review by TheNerdyHamster — COMMENTED** [review]
Looks good

**Review by TheNerdyHamster — APPROVED** [review]

**Pull request merged**